In [2]:
import pandas as pd
import numpy as np
from statsmodels.stats.weightstats import _tconfint_generic as t_stat
from scipy.stats import norm 
import math

# **Задача №1**

Известно, что генеральная совокупность распределена нормально 
со средним квадратическим отклонением, равным 16. 
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, 
если выборочная средняя M = 80, а объем выборки n = 256.

**Решение:**

1. Найдем среднее выборочное
2. Найдем доверительный интервал
   
   $$\overline{x} ± z_\frac{a}{2} * \frac{σ}{\sqrt{n}}$$

Дано:
$$\overline{x} = 80, σ = 16, n = 256$$ 
$$z_\frac{a}{2}(a = 0.05/2) = 1,96 $$

Итого:
$$80 ± (1,96 * \frac{16}{\sqrt{256}})$$

In [7]:
80 + (1.96 * 16/256**0.5)

81.96

In [8]:
80 - (1.96 * 16/256**0.5)

78.04

Ответ:

Доверительный интервал для оценки мат. ожидания генеральной совокупности: [78,04; 81,96]

# **Задача №2**

В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, 
получены опытные данные:
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, 
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это 
значение с доверительной вероятностью 0,95.

**Решение:**

Сигма совокупности неизвестна: 
$$\overline{x} ± t_\frac{a}{2} * \frac{σ}{\sqrt{n}}$$

1. Задаем массив
2. Ищем среднее арифметическое 
3. Ищем несмещенную дисперсию
4. Ищем т-критерий + доверительный интервал

In [11]:
X = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
mean_X = X.mean()
std_X = X.std(ddof=1)
mean_std_X = std_X / (np.sqrt(len(X)))
print(mean_X, std_X, mean_std_X)

6.590000000000001 0.4508017549014448 0.142556031868954


In [12]:
t_stat(mean_X, mean_std_X,len(X) - 1, 0.05, 'two-sided')

(6.267515851415713, 6.912484148584288)

**Ответ:**  доверительный интервал для истинного значения X : [6.267 ; 6.912]

# **Задача №3**

Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170

Рост матерей 178, 165, 165, 173, 168, 155, 160, 164, 178, 175

Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.

Интервальная оценка для разности средних арифметических ищнтся по формуле:

$$Δ ± t_\frac{a}{2} * S_Δ$$
где,
$$Δ(дельта) = \overline{X}_1 - \overline{X}_2$$

Объединенная оценка дисперсии: 
$$D = \frac{1}{2}(D_1+D_2)$$

Стандартная ошибка разности средних:
$$S_Δ = \sqrt{\frac{D}{n_1}+\frac{D}{n_2}} $$

In [18]:
moms= np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175]) 
dots = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])
import scipy.stats as stats 

In [14]:
mean_mothers = np.mean(moms) 
mean_daughters = np.mean(dots) 
print(f'Среднее арифметическое матерей - {mean_mothers}, дочерей {mean_daughters}')

Среднее арифметическое матерей - 168.1, дочерей 166.2


In [15]:
delta = round(mean_mothers - mean_daughters, 3)
# несмещенная дисперсия
D1 = round(np.var(moms, ddof=1), 3) 
D2 = round(np.var(dots, ddof=1), 3) 
D = (D1 + D2) / 2 
D

75.5835

In [20]:
SE = np.sqrt(D/len(moms) + D/len(dots))
t_tabul = stats.t.ppf(0.975, len(moms) + len(dots) - 2)
L = delta - t_tabul * SE
U = delta + t_tabul * SE
print(f"Доверительный интервал [{L}; {U}]")

Доверительный интервал [-6.268427040475464; 10.068427040475465]


**Ответ:** доверительный интервал для разности роста матерей и дочерей [-6.2684; 10.0684]